fabfile/utils.py

```
@@ -424,3 +429,80 @@ def show_time_deploy(td, show=False):
         print(yellow(time_deploy))
 
     return time_deploy
+
+
+class SupervisionHandler(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def stop_supervision(self, host, service, duration=None):
+        """
+        TODO
+        """
+        return
+
+    @abstractmethod
+    def start_supervision(self, host, service):
+        """
+        TODO
+        """
+        return
+
+
+class NagiosSupervisionHandler(SupervisionHandler):
+    def __init__(self, url, user, pwd):
+        self.url = url
+        self.user = user
+        self.pwd = pwd
+
+    def format_host(self, host):
+        """the nagio supervisor only wants a short host name"""
+        move_host = re.compile(r'@(.*?)\.')
+        return move_host.search(host).group(1)
+
+    def stop_supervision(self, host, service, duration=None):
+        start_dt = time.time()
+        start_str = datetime.datetime.fromtimestamp(start_dt).strftime('%d-%m-%Y %H:%M:%S')
+        if duration is None:
+            print("ERROR: 'duration' attribute cannot be a NoneType")
```

you need to raise an error there, we want to stop, not just skip the supervision monitoring